user/pages/03.commerce2/02.developer-guide/04.cart/04.cart-events/docs.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Cart Events
+taxonomy:
+    category: docs
+---
+
+Drupal Commerce's cart system provides a full range of events for your custom functionality and modules to react to.
+
+The events use Drupal's standard [Symphony based event system](https://www.drupal.org/docs/8/creating-custom-modules/subscribe-to-and-dispatch-events#s-event-systems-overview). So a solid understanding of that is useful. Events are defined in `\Drupal\commerce_cart\Event\CartEvents`.
+
+Events provided:
+- `CartEvents::CART_EMPTY`: Fired after emptying the cart.
+- `CartEvents::CART_ENTITY_ADD`: Fired after adding a purchaseable entity to the cart.
+- `CartEvents::CART_ORDER_ITEM_UPDATE`: Fired after an order item.
+- `CartEvents::CART_ORDER_ITEM_REMOVE`: Fired after removing an order item form the cart.
+- `CartEvents::CART_ITEM_COMPARISON_FIELDS`: Fired when altering the list of comparison fields.
```

Comment:
This probably needs a bit more explanation. What are "comparison fields"? Specifically, let's explain that this event can used to alter the "combine items" behavior for matching new items with existing items in the cart.